vispy/visuals/mesh.py

```diff
@@ -357,16 +424,77 @@ def _update_data(self):
                 normals = (1., 0., 0.)
 
             self.shared_program.vert['normal'] = normals
-            self.shared_program.vert['base_color'] = colors
+            self.shared_program.vert['color_transform'] = \
+                _build_color_transform(colors, self._cmap, self._clim_values)
+            if colors.ndim == 1:
+                self.shared_program.vert['base_color'] = colors
+            else:
+                self.shared_program.vert['base_color'] = VertexBuffer(colors)
 
             # Additional phong properties
-            self.shared_program.vert['light_dir'] = (10, 5, -5)
+            self.shared_program.vert['light_dir'] = self._light_dir
             self.shared_program.vert['light_color'] = (1.0, 1.0, 1.0, 1.0)
-            self.shared_program.vert['ambientk'] = (0.3, 0.3, 0.3, 1.0)
+            self.shared_program.vert['ambientk'] = \
+                self._ambient_light_color.rgba
+            self.shared_program.frag['shininess'] = self._shininess
 
         self._data_changed = False
 
     @property
+    def shininess(self):
+        """The shininess"""
+        return self._shininess
+
+    @shininess.setter
+    def shininess(self, shine):
+        """Set the shininess
+
+        Parameters
+        ----------
+        shine : float
+            The shininess to use.
+        """
+        self._shininess = float(shine)
+        self.mesh_data_changed()
+
+    @property
+    def ambient_light_color(self):
+        """The ambient light color"""
+        return self._ambient_light_color
+
+    @ambient_light_color.setter
+    def ambient_light_color(self, ambient):
+        """Set the ambient light
+
+        Parameters
+        ----------
+        color : instance of Color
+            The color to use.
+        """
+        self._ambient_light_color = Color(ambient)
+        self.mesh_data_changed()
+
+    @property
+    def light_dir(self):
+        """The light direction"""
+        return self._light_dir
+
+    @light_dir.setter
+    def light_dir(self, direction):
+        """Set the light direction
+
+        Parameters
+        ----------
+        direction : ndarray, shape (3,)
+            The light direction.
+        """
+        direction = np.array(direction, float).ravel()
+        if direction.size != 3 or not np.isfinite(direction).all():
+            raise ValueError('Invalid direction %s' % direction)
+        self._light_dir = tuple(direction)
+        self.mesh_data_changed()
+
```

Comment:
All of these other changes are merely to expose some parameters that were formerly hard-coded.